spring-cloud-gcp-autoconfigure/src/main/java/com/google/cloud/spring/autoconfigure/logging/StackdriverLoggingAutoConfiguration.java

```diff
@@ -44,7 +44,7 @@
  */
 @Configuration(proxyBeanMethods = false)
 @ConditionalOnClass({ HandlerInterceptorAdapter.class, LoggingAppender.class, TraceIdExtractor.class })
-@ConditionalOnMissingBean(type = "org.springframework.cloud.sleuth.autoconfig.SleuthProperties")
+@ConditionalOnMissingBean(name = "stackdriverTracingCustomizer")
```

Comment:
What is the purpose of this condition?
I'm also not sure what the purpose of the original condition you replaced was.

Reply:
I believe it's intended to turn off this config when Sleuth is present, but because the class we were previously using was changed to package private, we had to do something else.

Reply:
Logging should work when Sleuth is present. Since this config is triggered after `StackdriverTraceAutoConfiguration`, this condition will always be true. I think we should just remove it.

Reply:
It's not clear what the ramifications are when both Logging and Sleuth are on, but there were at least a few unit tests ensuring the proper beans were being configured, so I imagine this was here for a reason.

Logging _does_ work with Sleuth both turned on and off - I think the primary case is to ensure the `TraceIdExtractor` is _not_ enabled when Sleuth is turned on, since it looks like `StackdriverTracePropagation` (from `brave-propagation-stackdriver` library) handles the `x-cloud-trace-id` header. _I think_.

Reply:
I think you're right... we wanted to extract the trace ID and add it to log entries, even when Sleuth was not enabled. However, when Sleuth is enabled, we get the trace ID from MDC. Took me a while to remember that.